### PR TITLE
add csv encoding tests

### PR DESCRIPTION
**Pull request by KutuzovGregory**
* Windows-1251
* ISO 8859-1
add csv upload tests
cases:
* binary
* multipart
---
plain
tar.gz
gz
zip

### CODE REVIEW

File: src/test/java/com/axibase/tsd/api/method/csv/CSVUploadMethod.java [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+        return (url == null) ? null : new File(url.toURI());
+    }
+
+    public static boolean importParser(File configPath) {
+        Invocation.Builder builder = httpRootResource.path("/csv/configs/import").request();
+        MultiPart multiPart = new MultiPart();
+        FileDataBodyPart fileDataBodyPart
+                = new FileDataBodyPart("file", configPath, getMediaTypeFromFile(configPath));
+        multiPart.bodyPart(fileDataBodyPart);
+        Response response = builder.post(Entity.entity(multiPart, multiPart.getMediaType()));
```
Comment (unrealwork) [inline src/test/java/com/axibase/tsd/api/method/csv/CSVUploadMethod.java]: response is not closed

File: src/test/java/com/axibase/tsd/api/method/csv/CSVUploadTest.java [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+
+    private void checkMultipartFileUpload(String entityName, String metricName, File csvPath) throws Exception {
+        Registry.Entity.registerPrefix(entityName);
+        Registry.Metric.registerPrefix(metricName);
+
+        Response response = multipartCsvUpload(csvPath, PARSER_NAME);
+
+        assertEquals(response.getStatus(), OK.getStatusCode());
+
+        Thread.sleep(1000l);
```
Comment (Sild) [inline src/test/java/com/axibase/tsd/api/method/csv/CSVUploadTest.java]: please use upper case to long

File: src/test/java/com/axibase/tsd/api/model/message/Message.java [hunk 3]
```diff
@@ -1,10 +1,11 @@
 package com.axibase.tsd.api.model.message;
 
-import com.axibase.tsd.api.model.Model;
 import com.axibase.tsd.api.Registry;
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
 import com.fasterxml.jackson.annotation.JsonInclude;
 
 @JsonInclude(JsonInclude.Include.NON_NULL)
+@JsonIgnoreProperties(ignoreUnknown = true)
```
Comment (Sild) [inline src/test/java/com/axibase/tsd/api/model/message/Message.java]: Let's support full model instead
https://github.com/axibase/atsd-docs/blob/master/api/data/messages/query.md#fields-1

File: src/test/java/com/axibase/tsd/api/method/csv/ParserEncodingTest.java [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+    /*
+    * #2916
+    * */
+    @Test
+    public void testCsvCorrectTextEncodingISO8859_1() throws Exception {
+        String controlSequence = "¡¢£¤¥¦§¨©ª«¬\u00AD®¯°±²³´µ¶·¸¹º»¼½¾¿ÀÁÂÃÄÅÆÇÈÉÊËÌÍÎÏÐÑÒÓÔÕÖ×ØÙÚÛÜÝÞßàáâãäåæçèéêëìíîïðñòóôõö÷øùúûüýþÿ";
+        String entityName = ENTITY_PREFIX+"-1";
+        File csvPath = resolvePath(RESOURCE_DIR + File.separator + name.getMethodName() + ".csv");
+
+        checkCsvCorrectTextEncoding(controlSequence, entityName, csvPath, "ISO-8859-1");
```
Comment (Sild) [inline src/test/java/com/axibase/tsd/api/method/csv/ParserEncodingTest.java]: Let move out charset to constant?

File: src/test/java/com/axibase/tsd/api/method/csv/ParserEncodingTest.java [hunk 5]
```diff
@@ -0,0 +1,78 @@
+package com.axibase.tsd.api.method.csv;
+
+import com.axibase.tsd.api.Registry;
+import com.axibase.tsd.api.Util;
+import com.axibase.tsd.api.method.message.MessageMethod;
+import com.axibase.tsd.api.model.message.Message;
+import com.axibase.tsd.api.model.message.MessageQuery;
+import org.json.simple.parser.ParseException;
```
Comment (Sild) [inline src/test/java/com/axibase/tsd/api/method/csv/ParserEncodingTest.java]: Where did you find this lib?

File: src/test/java/com/axibase/tsd/api/method/csv/ParserEncodingTest.java [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
+    @Test
+    public void testCsvCorrectTextEncodingWindows1251() throws Exception {
+        String controlSequence = "йцукенгшщзхъфывапролджэёячсмитьбюЙЦУКЕНГШЩЗХЪФЫВАПРОЛДЖЭЁЯЧСМИТЬБЮ";
+        String entityName = ENTITY_PREFIX+"-2";
+        File csvPath = resolvePath(RESOURCE_DIR + File.separator + name.getMethodName() + ".csv");
+
+        checkCsvCorrectTextEncoding(controlSequence, entityName, csvPath, "Windows-1251");
+    }
+
+    private void checkCsvCorrectTextEncoding(String controlSequence, String entityName, File csvPath, String textEncoding) throws InterruptedException, IOException, ParseException {
```
Comment (Sild) [inline src/test/java/com/axibase/tsd/api/method/csv/ParserEncodingTest.java]: ParseException from `json.simple` library is redundant

Comment (Sild) [inline src/test/java/com/axibase/tsd/api/method/csv/ParserEncodingTest.java]: end review

File: src/test/java/com/axibase/tsd/api/model/message/Message.java [hunk 7]
```diff
@@ -90,4 +93,13 @@ public Boolean getPersist() {
     public void setPersist(Boolean persist) {
         this.persist = persist;
     }
+
+    @JsonInclude(JsonInclude.Include.NON_EMPTY)
```
Comment (Sild) [inline src/test/java/com/axibase/tsd/api/model/message/Message.java]: Do you sure it's necessary?

**Comment by Sild** [comment]
OK

**Comment by unrealwork** [comment]
Ok

**Comment by rmakulov** [comment]
Ok

**Pull request merged**